bluemix/src/main/java/com/ibm/watson/SpeechToText.java

```diff
@@ -43,120 +39,157 @@ public String speechToText(byte[] incomingBinaryPOSTData) {
     */
         HttpURLConnection conn = null;
         String serviceURL = "";
-           
+
         try {
-	    if (incomingBinaryPOSTData.length == 0) return XML_DEC + "<results>" + STATUS_OK + "<transcript />" + "</results>";
+            if (incomingBinaryPOSTData.length == 0) {
+                return XML_DEC + "<results>" + STATUS_OK + "<transcript />" + "</results>";
+            }
 
             // 'VCAP_SERVICES' contains all the details of services bound to this application
-	    JsonReader jsonReader = Json.createReader(new StringReader(System.getenv("VCAP_SERVICES")));
-	    JsonObject serviceInfo = jsonReader.readObject();
-	    jsonReader.close();
+            JsonObject credentials = readJsonObject(new StringReader(System.getenv("VCAP_SERVICES")))
+                                    .getJsonArray(SERVICE_NAME)
+                                    .getJsonObject(0)
+                                    .getJsonObject("credentials");
 
-            // Get the Service Credentials for SpeechToText
-	    JsonObject credentials = serviceInfo.getJsonArray(SERVICE_NAME)
-		.getJsonObject(0).getJsonObject("credentials");
 
-            try {
-                serviceURL = credentials.getString("url");
-            } catch (Exception e) {}
-            // If we didn't find a URL for the service in VCAP_SERVICES,
-            // use the original.
+
+            serviceURL = credentials.getString("url");
+
```

Comment:
Nested exceptions can be an indicator that another pattern is possible.

```java
void doSomething() {
  try {
    Result result;
    try {
     result = somethingScary();
    } catch(Exception e) {
      handleException(e);
    }
    if(result == null) { handleFailedResult(); }
    doSomethingWithResult(result);
  } catch(Exception e) {
      handleLargerException(e);
  }
}
```

could be rewritten as

```java
void doSomething() {
  Result result = getResult();
  ensureSaneResult(result);
  safelyDoSomething(result);
}

void ensureSaneResult(Result result) {
  if(result == null) { handleFailedResult(); }
}

Result getResult() {
  Result result;
  try {
   result = somethingScary();
  } catch(Exception e) {
    handleException(e);
  }
  // could put the ensureSaneResult() sanity check here, too
  return result;
}

void safelyDoSomething(Result result) {
  try {  
    doSomethingWithResult(result);
  } catch(Exception e) {
      handleLargerException(e);
  }
}
```

It's more verbose but its meaning is so much clearer.

In Scala, it might be even cleaner, if the underlying actionable methods played along nicely and used Option and whatnot.

```scala
def doSomething() = {
  Try(somethingScary).map(doSomethingWithResult).recover(handleException)
}
```